libraries/state_history/include/eosio/state_history/serialization.hpp

```diff
@@ -356,7 +356,7 @@ datastream<ST>& operator<<(datastream<ST>& ds, const history_serial_wrapper<eosi
 
 template <typename ST>
 datastream<ST>& operator<<(datastream<ST>& ds, const history_serial_wrapper<eosio::chain::shared_key_weight>& obj) {
-   fc::raw::pack(ds, as_type<eosio::chain::public_key_type>(obj.obj.key));
+   fc::raw::pack(ds, as_type<eosio::chain::shared_public_key>(obj.obj.key));
```

Comment:
I'm pretty sure that we need this change: https://github.com/EOSIO/eos/pull/9088

Reply:
yes. fixed.